### PR TITLE
Implemented ability to detect concatinated strings in lexer

### PR DESCRIPTION
**Pull request by lapistano**

### CODE REVIEW

File: test/Parser/ParserUtilTest.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public function testGetStringLogicExpectingException()
+    {
+        $typeNotString = '([],foo)';
+
+        $lexer = new Lexer();
+        $lexer->setInput($typeNotString);
+
+        ParserUtil::parseStart($lexer);
+
+        $this->setExpectedException('\LogicException');
```
Comment (hairmare) [inline test/Parser/ParserUtilTest.php:63]: When I see the test I realize we should work on #17 and create proper `RqlBlahException` classes...

Comment (hairmare) [inline test/Parser/ParserUtilTest.php:63]: Also, could you do this as an annotation? I prefer that syntax where possible...

Comment (hairmare) [inline test/Parser/ParserUtilTest.php:63]: ie. `@expectedException \LogicException`

Comment (lapistano) [inline test/Parser/ParserUtilTest.php:63]: The way it is now any `\LogicException` thrown before it is explicitly registered by the setter will case the test to fail. If the annotation is used, any thrown `\LogicException` will result in a successful test.

Comment (hairmare) [inline test/Parser/ParserUtilTest.php:63]: Ah ok, no worries then. Any idea on how we could do something about duplication as per scrutinizer? Should we ignore it?

Comment (lapistano) [inline test/Parser/ParserUtilTest.php:63]: jo

**Pull request merged**